base/dict.jl

```diff
@@ -668,23 +668,68 @@ function skip_deleted_floor!(h::Dict)
     idx
 end
 
-@propagate_inbounds _iterate(t::Dict{K,V}, i) where {K,V} = i > length(t.vals) ? nothing : (Pair{K,V}(t.keys[i],t.vals[i]), i+1)
-@propagate_inbounds function iterate(t::Dict)
-    _iterate(t, skip_deleted_floor!(t))
+@inline function _iterate(h::Dict)
+    (isempty(h) || length(h.slots)<8) && return nothing
```

Comment:
isn't the default size of `slots` in an empty `Dict` 16, so why 8?

Reply:
I was checking for `length(h.slots) < 8` because I am about to read 8 bytes. But you're right, the check is redundant.